codec-memcache/src/test/java/io/netty/handler/codec/memcache/binary/DefaultFullBinaryMemcacheRequestTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2013 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.memcache.binary;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.buffer.Unpooled;
+import io.netty.util.CharsetUtil;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class DefaultFullBinaryMemcacheRequestTest {
+
+    private DefaultFullBinaryMemcacheRequest request;
+
+    @Before
+    public void setUp() {
+        request = new DefaultFullBinaryMemcacheRequest(
+                Unpooled.copiedBuffer("key", CharsetUtil.UTF_8),
+                Unpooled.wrappedBuffer(new byte[]{1, 3, 4, 9}),
+                Unpooled.copiedBuffer("some value", CharsetUtil.UTF_8));
+        request.setReserved((short) 534);
+        request.setMagic((byte) 0x03);
+        request.setOpcode((byte) 0x02);
+        request.setKeyLength((short) 32);
+        request.setExtrasLength((byte) 34);
+        request.setDataType((byte) 43);
+        request.setTotalBodyLength(345);
+        request.setOpaque(3);
+        request.setCas(345345L);
+    }
+
+    @Test
+    public void fullCopy() {
+        assertCopy(request, request.content(), request.copy());
+    }
+
+    @Test
+    public void fullDuplicate() {
+        assertCopy(request, request.content(), request.duplicate());
+    }
+
+    @Test
+    public void fullReplace() {
+        ByteBuf newContent = Unpooled.copiedBuffer("new value", CharsetUtil.UTF_8);
+        assertCopy(request, newContent, request.replace(newContent));
+    }
+
+    private void assertCopy(FullBinaryMemcacheRequest expected, ByteBuf expectedContent,
+                            FullBinaryMemcacheRequest actual) {
+        assertNotSame(expected, actual);
+
+        assertTrue(ByteBufUtil.equals(expected.key(), actual.key()));
```

Comment:
just use `assertEquals(expected.key(), actual.key()` .... same for the other lines

Reply:
Addressed